### PR TITLE
i#2138: fix crash in Win10 x64 clients requesting extra syscall inter…

### LINKED ISSUE #2138: DrCov crash on Windows 10 x64 when tracing Notepad File Open Dialog
I tested this on both 6.2.0 and latest git today, I'm posting crash info from the 6.2.0 build since you have those binaries. This occurs when doing File->Open in Notepad. The dialog partially renders and then crashes. 

$ bin64/drrun.exe -t drcov -- /c/windows/notepad.exe
<Application C:\windows\notepad.exe (12392).  DrCov internal crash at PC 0x000000007109dbe4.  Please report this at http://dynamorio.org/issues.  Program aborted.
0xc0000005 0x00000000 0x000000007109dbe4 0x000000007109dbe4 0x0000000000000000 0xffffffffffffffb1
Base: 0x0000000071000000
Registers: eax=0x00007ff6598f2e01 ebx=0x0000000000000000 ecx=0x0000000000000000 edx=0x0000000000000018
        esi=0x00000000710654c6 edi=0x00007ff6598f2e01 esp=0x00007ff65996edf0 ebp=0x0000000000000001
        r8 =0x0000000000000018 r9 =0x00007ff65996e6d8 r10=0x0000000000000000 r11=0x00007ff65996ede0
        r12=0x0000000000000004 r13=0x00007ff659952080 r14=0x00000000710ecb70 r15=0x0000000000000001
        eflags=0x00007ff600010206
version 6.2.0, build 2
-no_dynamic_options -client_lib 'C:\Code\disas_benchmark\DynamoRIO-Windows-6.2.0-2\tools\lib64\release\drcov.dll;0;' -code_api -probe_api -stack_size 56K -nop_initial_bblock -max_elide_jmp 0 -max_elide_call 0 -no_inline_ignored_syscalls -native_exec_default_list '' -no_native_exec_managed_code -no_indcall2direct -pad_jmp>

### PR DESCRIPTION
**Pull request by shawndenbow**
…ception

Extend the size of the 64-bit interception buffer, as it is exceeded by
a client requesting extra syscalls be intercepted while native.

Increase the MAX_CONTEXT_LEN, as it was observed higher on Windows 10 x64

Fixes #2138

### CODE REVIEW

File: core/win32/os_private.h [hunk 1]
```diff
@@ -331,7 +331,7 @@ os_rename_file_in_directory(IN HANDLE rootdir,
 /* thread-shared only needs 4 pages on 32-bit but -thread_private needs 5
  * in case we hook the image entry on an early cbret
  */
-#define INTERCEPTION_CODE_SIZE IF_X64_ELSE(8*4096,7*4096)
+#define INTERCEPTION_CODE_SIZE IF_X64_ELSE(9*4096,7*4096)
```
Comment (derekbruening) [inline core/win32/os_private.h]: Let's add a comment explaining that we need extra space for dr_syscall_intercept_natively.

File: core/win32/os_private.h [hunk 2]
```diff
@@ -461,11 +461,11 @@ extern uint context_xstate;
 
 #define XSTATE_HEADER_SIZE 0x40  /* 512 bits */
 #define YMMH_AREA(ymmh_area, i) (((dr_xmm_t*)ymmh_area)[i])
-#define MAX_CONTEXT_64_SIZE     0x680 /* 0x66f from win-7 sp1 */
+#define MAX_CONTEXT_64_SIZE     0x6ef /* 0x66f from win-7 sp1 */
```
Comment (derekbruening) [inline core/win32/os_private.h]: Update comment, maybe to just "/* as observed on win10-x64 */" or sthg?

File: core/win32/os_private.h [hunk 3]
```diff
@@ -461,11 +461,11 @@ extern uint context_xstate;
...
 #ifdef X64
 # define MAX_CONTEXT_SIZE       MAX_CONTEXT_64_SIZE
 #else
-# define MAX_CONTEXT_SIZE       0x480 /* 0x463 from win-7 sp1 */
+# define MAX_CONTEXT_SIZE       0x4e3 /* 0x463 from win-7 sp1 */
```
Comment (derekbruening) [inline core/win32/os_private.h]: Ditto

File: core/win32/os_private.h (hunk 1 above)
Comment (shawndenbow) [inline core/win32/os_private.h]: Done.

File: core/win32/os_private.h (hunk 2 above)
Comment (shawndenbow) [inline core/win32/os_private.h]: Done.

File: core/win32/os_private.h (hunk 3 above)
Comment (shawndenbow) [inline core/win32/os_private.h]: Done.

**Comment by derekbruening** [comment]
I'm going to go ahead and update vs master and cancel the prior CI runs for the new ones.

**Comment by sssinghsyr** [comment]
Hi, 
I am struggling with building package for the latest code with this issue fixed. 
Could you please provide me the copy of your local release package for this version?

**Pull request merged**